src/functions/template-tags/general.php

```diff
@@ -1518,48 +1527,60 @@ function tribe_events_get_the_excerpt( $post = null, $allowed_html = null, $skip
 		 */
 		$remove_shortcodes = apply_filters( 'tribe_events_excerpt_shortcode_removal', true );
 
-		// Get the Excerpt or content based on what is available
-		$excerpt = has_excerpt( $post->ID ) ? $post->post_excerpt : $post->post_content;
+		$cache_excerpts_key = implode( ':', [
+			$post->ID,
+			$skip_postdata_manipulation,
+			$allow_shortcodes,
+			$remove_shortcodes,
+			json_encode( $allowed_html )
+		] );
+
+		if ( ! isset( $cache_excerpts[ $cache_excerpts_key ] ) ) {
+			// Get the Excerpt or content based on what is available
+			$excerpt = has_excerpt( $post->ID ) ? $post->post_excerpt : $post->post_content;
+
+			// If shortcode filter is enabled let's process them
+			if ( $allow_shortcodes ) {
+				$excerpt = do_shortcode( $excerpt );
+			}
 
-		// If shortcode filter is enabled let's process them
-		if ( $allow_shortcodes ) {
-			$excerpt = do_shortcode( $excerpt );
-		}
+			// Remove all shortcode Content before removing HTML
+			if ( $remove_shortcodes ) {
+				$excerpt = preg_replace( '#\[.+\]#U', '', $excerpt );
+			}
 
-		// Remove all shortcode Content before removing HTML
-		if ( $remove_shortcodes ) {
-			$excerpt = preg_replace( '#\[.+\]#U', '', $excerpt );
-		}
+			// Remove "all" HTML based on what is allowed
+			$excerpt = wp_kses( $excerpt, $allowed_html );
 
-		// Remove "all" HTML based on what is allowed
-		$excerpt = wp_kses( $excerpt, $allowed_html );
+			if ( ! has_excerpt( $post->ID ) ) {
+				// Temporarily alter the global post in preparation for our filters.
+				$global_post = isset( $GLOBALS['post'] ) ? $GLOBALS['post'] : null;
+				$GLOBALS['post'] = $post;
 
-		if ( ! has_excerpt( $post->ID ) ) {
-			// Temporarily alter the global post in preparation for our filters.
-			$global_post = isset( $GLOBALS['post'] ) ? $GLOBALS['post'] : null;
-			$GLOBALS['post'] = $post;
+				// We will only trim Excerpt if it comes from Post Content
 
-			// We will only trim Excerpt if it comes from Post Content
+				/**
+				 * Filter the number of words in an excerpt.
+				 *
+				 * @param int $number The number of words. Default 55.
+				 */
+				$excerpt_length = apply_filters( 'excerpt_length', 55 );
 
-			/**
-			 * Filter the number of words in an excerpt.
-			 *
-			 * @param int $number The number of words. Default 55.
-			 */
-			$excerpt_length = apply_filters( 'excerpt_length', 55 );
+				/**
+				 * Filter the string in the "more" link displayed after a trimmed excerpt.
+				 *
+				 * @param string $more_string The string shown within the more link.
+				 */
+				$excerpt_more = apply_filters( 'excerpt_more', ' [&hellip;]' );
 
-			/**
-			 * Filter the string in the "more" link displayed after a trimmed excerpt.
-			 *
-			 * @param string $more_string The string shown within the more link.
-			 */
-			$excerpt_more = apply_filters( 'excerpt_more', ' [&hellip;]' );
+				// Now we actually trim it
```

Comment:
:no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: b98f55bbd1ecc55994ea56c13000d202</sub>